src/riemannian_diff.jl

```diff
@@ -363,7 +363,7 @@ function riemannian_Hessian(M::AbstractManifold, p, eG, eH, X)
     return Y
 end
 function riemannian_Hessian!(M::AbstractManifold, Y, p, eG, eH, X)
-    project!(M, Y, p, eH) #first term
-    Y .= Y + Weingarten(M, p, X, eG - project(M, p, eG))
+    project!(M, Y, p, eH) #first term - project the Euclidean Hessian
+    copyto!(M, p, Y, Y + Weingarten(M, p, X, eG - project(M, p, eG)))
```

Comment:
Broadcasting should work fine here, even with `Y.+= Weigarten(...)`.

Reply:
Ok, will change that (I am never so sure about these).

Reply:
Ok :)